File: src/main/java/it/contrader/dto/CommitDTO.java
package it.contrader.dto;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIdentityInfo(generator=ObjectIdGenerators.PropertyGenerator.class, property="id")

public class CommitDTO {

	private Long id;
	private String date;
	private int stars;
	private int fork;
	private int watcher;
	private int collaborators;
	private int number;
	
	private CoinDTO coinDTO;
	

}

File: src/dto/commitdto.ts
import { CoinDTO } from './coindto';


export class CommitDTO {


    id: number ;

    date: string;

    stars: number;
    
    fork: number;

    watcher: number;

    collaborators: number;
    
    number: number;

    coindto: CoinDTO;
}